v3/src/test/setupTests.ts

```diff
@@ -17,3 +18,50 @@ declare global {
 }
 global.assertIsDefined = assertIsDefined
 global.jestSpyConsole = jestSpyConsole
+
+let message = () => ""
+
+function removeIdsReplacer(key: string, value: any) {
+  return key === "id" ? undefined : value
+}
+
+// for use with lodash's isEqualWith function to implement toEqualWithUniqueIds custom jest matcher
+const withoutIdsCustomizer = (rec: any, exp: any) => {
+  if (Array.isArray(rec) && Array.isArray(exp) && (rec?.length !== exp?.length)) {
+    message = () =>
+      `array lengths must be equal\n` +
+      `  Expected: ${exp.length}\n` +
+      `  Received: ${rec.length}`
+    return false
+  }
+  const _rec = JSON.parse(JSON.stringify(rec, removeIdsReplacer))
+  const _exp = JSON.parse(JSON.stringify(exp, removeIdsReplacer))
+  if (!isEqual(_rec, _exp)) {
+    message = () =>
+      `object values must match:\n` +
+      `  Expected: ${JSON.stringify(_exp)}\n` +
+      `  Received: ${JSON.stringify(_rec)}`
+  }
+  return true
+}
+
+// custom jest matcher for testing that copied objects are identical to the originals
+// except for the object ids which should all be different.
+expect.extend({
+  toEqualWithoutIds(received: any, expected: any) {
+    message = () => ""
+    const pass = isEqualWith(received, expected, withoutIdsCustomizer)
+    if (!pass && !message()) {
+      message = () => "values differed outside of geometry objects"
```

Comment:
What does "geometry objects" mean here? I think I understand the general meaning of the message -- that there was a mismatch of something outside of the checks in withoutIdsCustomizer -- but not the "geometry objects" part. I'm probably just ignorant of the term, sorry.

Reply:
Holdover from CLUE -- I'll adjust the message.